osh/hub/scripts/osh-worker-manager

```diff
@@ -0,0 +1,203 @@
+
+
+def create_worker(ip_address):
+    # Worker name should be set to hostname of the worker
+    worker_name = get_hostname_from_ip(ip_address)
+    kobo_worker = create_worker_in_kobo(worker_name)
+    start_worker(kobo_worker)
+
+
+def delete_worker(ip_address):
+    from kobo.hub import models
+
+    # Worker name should be set to hostname of the worker
+    worker_name = get_hostname_from_ip(ip_address)
+    kobo_worker = models.Worker.objects.get(name=worker_name)
+    kobo_worker.delete()
+
+
+def workers_needed():
+    import kobo.hub.models
+    from django.conf import settings
+
+    # Tasks in opened or assigned state
+    running_tasks = kobo.hub.models.Task.objects.running().count()
+
+    # Tasks waiting to be assigned
+    free_tasks = kobo.hub.models.Task.objects.free().count()
+
+    running_and_free_tasks = running_tasks + free_tasks
+
+    print(min(running_and_free_tasks, settings.MAX_SINGLE_USE_WORKERS))
+
+
+def check_finished(ip_address):
+    import sys
+
+    from kobo.hub import models
+
+    # Worker name should be set to hostname of the worker
+    worker_name = get_hostname_from_ip(ip_address)
+    kobo_worker = models.Worker.objects.get(name=worker_name)
+
+    # TODO: Check for stale workers that have max_load above 0, but do not have arch
+    # or channel set. They would never be used.
+
+    # TODO: Shall we perform further health checks here? For example, if `osh-worker`
+    # service is running on the worker machine?
+    if kobo_worker.max_load == 0 and kobo_worker.current_load == 0:
+        sys.exit(0)
+
+    print(f"{worker_name} is either running a task or new.", file=sys.stderr)
+
+    sys.exit(1)
+
+
+def main():
+    import django
+    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "osh.hub.settings")
+    django.setup()
+
+    import argparse
+    parser = argparse.ArgumentParser()
+
+    # These options are used by resalloc-agent-spawner.
+    # https://github.com/praiskup/resalloc/blob/151bd591ca2426b8a6b99253d88761c6e4be9f8c/config/agent-spawner/config.yaml
+    parser.add_argument("--create-worker", type=str, help="Try to create worker with specified hostname.")
+    parser.add_argument("--delete-worker", help="Delete worker with specified hostname (if its ideal)")
+    parser.add_argument("--workers-needed", action="store_true", help="Count number of queued task and return number of workers needed.")
+    parser.add_argument("--check-finished", type=str, help="Check if the worker should be removed.")
+
+    args = parser.parse_args()
+
+    if args.create_worker:
+        print(f"Creating {args.create_worker}")
+        ip_address = args.create_worker
+        create_worker(ip_address)
+
+    if args.delete_worker:
+        print(f"Deleting {args.delete_worker}")
+        ip_address = args.delete_worker
+        delete_worker(ip_address)
+
+    if args.workers_needed:
+        workers_needed()
+
+    if args.check_finished:
+        check_finished(args.check_finished)
```

Comment:
For some commands, you use the  redundant `ip_address` variable, for some you do not.  I vote to use the `args.<name>` member directly.